--- ../database/database.mysql~	2005-03-23 22:36:41.000000000 +0200
+++ ../database/database.mysql	2005-05-18 18:41:49.000000000 +0300
@@ -775,19 +775,19 @@
 -- Insert some default values
 --
 
-INSERT INTO system VALUES ('modules/block.module','block','module','',1,0,0);
-INSERT INTO system VALUES ('modules/comment.module','comment','module','',1,0,0);
-INSERT INTO system VALUES ('modules/filter.module','filter','module','',1,0,0);
-INSERT INTO system VALUES ('modules/help.module','help','module','',1,0,0);
-INSERT INTO system VALUES ('modules/node.module','node','module','',1,0,0);
-INSERT INTO system VALUES ('modules/page.module','page','module','',1,0,0);
-INSERT INTO system VALUES ('modules/story.module','story','module','',1,0,0);
-INSERT INTO system VALUES ('modules/system.module','system','module','',1,0,0);
-INSERT INTO system VALUES ('modules/taxonomy.module','taxonomy','module','',1,0,0);
-INSERT INTO system VALUES ('modules/user.module','user','module','',1,0,0);
-INSERT INTO system VALUES ('modules/watchdog.module','watchdog','module','',1,0,0);
-INSERT INTO system VALUES ('themes/bluemarine/xtemplate.xtmpl','bluemarine','theme','themes/engines/xtemplate/xtemplate.engine',1,0,0);
-INSERT INTO system VALUES ('themes/engines/xtemplate/xtemplate.engine','xtemplate','theme_engine','',1,0,0);
+INSERT INTO system VALUES ('/usr/share/drupal/modules/block.module','block','module','',1,0,0);
+INSERT INTO system VALUES ('/usr/share/drupal/modules/comment.module','comment','module','',1,0,0);
+INSERT INTO system VALUES ('/usr/share/drupal/modules/filter.module','filter','module','',1,0,0);
+INSERT INTO system VALUES ('/usr/share/drupal/modules/help.module','help','module','',1,0,0);
+INSERT INTO system VALUES ('/usr/share/drupal/modules/node.module','node','module','',1,0,0);
+INSERT INTO system VALUES ('/usr/share/drupal/modules/page.module','page','module','',1,0,0);
+INSERT INTO system VALUES ('/usr/share/drupal/modules/story.module','story','module','',1,0,0);
+INSERT INTO system VALUES ('/usr/share/drupal/modules/system.module','system','module','',1,0,0);
+INSERT INTO system VALUES ('/usr/share/drupal/modules/taxonomy.module','taxonomy','module','',1,0,0);
+INSERT INTO system VALUES ('/usr/share/drupal/modules/user.module','user','module','',1,0,0);
+INSERT INTO system VALUES ('/usr/share/drupal/modules/watchdog.module','watchdog','module','',1,0,0);
+INSERT INTO system VALUES ('themes/bluemarine/xtemplate.xtmpl','bluemarine','theme','themes/engines/xtemplate/xtemplate.engine',1,0,0);
+INSERT INTO system VALUES ('themes/engines/xtemplate/xtemplate.engine','xtemplate','theme_engine','',1,0,0);
 INSERT INTO users (uid, name, mail) VALUES ('0', '', '');
 INSERT INTO users_roles (uid, rid) VALUES (0, 1);
 
