--- drupal-5.0/includes/bootstrap.inc~	2007-01-26 13:05:53.697188263 +0200
+++ drupal-5.0/includes/bootstrap.inc	2007-01-26 13:06:43.528313680 +0200
@@ -199,7 +199,7 @@
     return $conf;
   }
 
-  $confdir = 'sites';
+  $confdir = '/etc/webapps/drupal/sites';
   $uri = explode('/', $_SERVER['PHP_SELF'] ? $_SERVER['PHP_SELF'] : $_SERVER['SCRIPT_FILENAME']);
   $server = explode('.', implode('.', array_reverse(explode(':', rtrim($_SERVER['HTTP_HOST'], '.')))));
   for ($i = count($uri) - 1; $i > 0; $i--) {
