--- drupal-4.6.5/includes/database.mysql.inc~	2005-12-20 12:05:30.000000000 +0200
+++ drupal-4.6.5/includes/database.mysql.inc	2005-12-20 12:06:30.000000000 +0200
@@ -29,7 +29,7 @@
   }
 
   // For replication setups, we will assume that dieing is bad 
-  $connection = mysql_connect($url['host'], $url['user'], $url['pass'], TRUE);
+  $connection = mysql_connect($url['host'], $url['user'], (isset($url['pass']) ? $url['pass'] : null), TRUE);
   if (!is_resource($connection) && $critical) {
     die(mysql_error());
   }
