--- ./includes/bootstrap.inc	2005-10-11 10:02:05.000000000 +0300
+++ /tmp/bootstrap.inc	2005-10-11 10:01:51.000000000 +0300
@@ -50,7 +50,7 @@
 
   $confdir = '/etc/drupal/sites';
   $uri = explode('/', $_SERVER['PHP_SELF']);
-  $server = explode('.', rtrim($_SERVER['HTTP_HOST'], '.'));
+  $server = explode('.', rtrim(@$_SERVER['HTTP_HOST'], '.'));
   for ($i = count($uri) - 1; $i > 0; $i--) {
     for ($j = count($server); $j > 0; $j--) {
       $dir = implode('.', array_slice($server, -$j)) . implode('.', array_slice($uri, 0, $i));
--- ./includes/session.inc	2005-08-10 23:43:03.000000000 +0300
+++ /tmp/session.inc	2005-10-11 10:03:38.000000000 +0300
@@ -25,7 +25,7 @@
   $result = db_query_range("SELECT u.*, s.* FROM {users} u INNER JOIN {sessions} s ON u.uid = s.uid WHERE s.sid = '%s' AND u.status < 3", $key, 0, 1);
 
   if (!db_num_rows($result)) {
-    db_query("INSERT INTO {sessions} (sid, uid, hostname, timestamp) VALUES ('%s', 0, '%s', %d)", $key, $_SERVER["REMOTE_ADDR"], time());
+    db_query("INSERT INTO {sessions} (sid, uid, hostname, timestamp) VALUES ('%s', 0, '%s', %d)", $key, @$_SERVER["REMOTE_ADDR"], time());
     $result = db_query("SELECT u.* FROM {users} u WHERE u.uid = 0");
   }
 
--- ./cron.php	2005-12-09 20:13:27.000000000 +0200
+++ drupal-4.6.11/cron.php	2007-01-30 18:02:41.911795649 +0200
@@ -1,3 +1,4 @@
+#!/usr/bin/php
 <?php
 // $Id$
 
@@ -6,6 +7,13 @@
  * Handles incoming requests to fire off regularly-scheduled tasks (cron jobs).
  */
 
+// If not in 'safe mode', increase the memory limit
+if (!ini_get('safe_mode')) {
+  ini_set('memory_limit', '20M');
+}
+
+chdir(dirname(__FILE__));
+
 include_once 'includes/bootstrap.inc';
 include_once 'includes/common.inc' ;
 
