--- drupal-5.0/includes/theme.inc~	2007-01-26 13:09:14.801730081 +0200
+++ drupal-5.0/includes/theme.inc	2007-01-26 13:09:52.242575644 +0200
@@ -65,6 +65,7 @@
     // File is a template/theme
     // Load its CSS, if it exists
     if (file_exists($stylesheet = dirname($themes[$theme]->filename) .'/style.css')) {
+    if (file_exists('htdocs/' . ($stylesheet = dirname($themes[$theme]->filename) .'/style.css'))) {
       drupal_add_css($stylesheet, 'theme');
     }
   }
