diff -uNdr drupal-5.7.old1/includes/bootstrap.inc drupal-5.7/includes/bootstrap.inc
--- drupal-5.7.old1/includes/bootstrap.inc	2008-02-16 16:02:23.000000000 +0100
+++ drupal-5.7/includes/bootstrap.inc	2008-02-16 16:06:30.000000000 +0100
@@ -199,7 +199,7 @@
     return $conf;
   }
 
-  $confdir = 'sites';
+  $confdir = '/etc/webapps/drupal/sites';
   $uri = explode('/', $_SERVER['SCRIPT_NAME'] ? $_SERVER['SCRIPT_NAME'] : $_SERVER['SCRIPT_FILENAME']);
   $server = explode('.', implode('.', array_reverse(explode(':', rtrim(@$_SERVER['HTTP_HOST'], '.')))));
   for ($i = count($uri) - 1; $i > 0; $i--) {
