--- ./themes/engines/xtemplate/xtemplate.engine~	2005-05-18 19:00:15.397473761 +0300
+++ ./themes/engines/xtemplate/xtemplate.engine	2005-05-18 19:00:25.785782566 +0300
@@ -14,7 +14,7 @@
   }
   $GLOBALS["xtemplate"] = new StdClass();
   $dir = dirname($template->filename);
-  $GLOBALS['xtemplate']->template = new XTemplate(basename($template->filename), $dir);
+  $GLOBALS['xtemplate']->template = new XTemplate(basename($template->filename), "/usr/share/drupal/$dir");
   $GLOBALS['xtemplate']->template->assign(array('directory' => $dir));
   $GLOBALS['xtemplate']->template->SetNullBlock(' ');  // '' doesn't work!
 }
--- ../includes/theme.inc~	2005-05-18 18:57:36.196386437 +0300
+++ ../includes/theme.inc	2005-05-18 18:57:57.045993344 +0300
@@ -114,7 +114,7 @@
     $list = array();
     $result = db_query("SELECT * FROM {system} WHERE type = 'theme' ORDER BY name");
     while ($theme = db_fetch_object($result)) {
-      if (file_exists($theme->filename)) {
+      if (file_exists("/usr/share/drupal/$theme->filename")) {
         $list[$theme->name] = $theme;
       }
     }
